Website: update docsy theme

deining opened the pull request:
The website uses the [docsy](https://docsy.dev) theme. This PR converts this theme from git submodule to hugo module.
It also brings the theme to latest released version [v0,.2.0](https://github.com/google/docsy/releases/tag/v0.2.0).

dominikh commented on website/go.mod:
```diff
@@ -10,9 +10,13 @@ require (
 )
 
 require (
+	github.com/google/docsy v0.2.0 // indirect
```
What will prevent `go mod tidy` from removing this dependency? It's not used by anything that Go can see.

deining replied:
Good point. This is an issue indeed, please note that there is a related open [issue](https://github.com/gohugoio/hugo/issues/6115) in the hugo project that addresses this topic.
Since go 1.14, there is a [-modfile](https://tip.golang.org/doc/go1.14#go-flags) flag that instructs the go command to use an alternate `go.mod` file. Could this be a solution?

dominikh commented on website/go.mod:
```diff
@@ -10,9 +10,13 @@ require (
+)
+
+require (
+	github.com/google/docsy v0.2.0 // indirect
```
Duplicate entry

deining replied:
Thanks for poining that out. This is now fixed.

dominikh commented on website/i18n/en.toml:
```diff
@@ -1,3 +0,0 @@
-# Fix a typo in Docsy (although we don't currently use the print support)
```
Was this fixed upstream?

deining replied:
Yes, via this [PR](https://github.com/google/docsy/pull/760).

dominikh commented on website/config.toml:
```diff
@@ -1,12 +1,24 @@
 baseURL = "https://staticcheck.io"
 languageCode = 'en-us'
 title = 'Staticcheck'
-theme = 'docsy'
```
Does the theme get used automagically, just because we're importing the module?

deining replied:
No. This section, which was added at the end of `config.toml` imports the modules.

```
[module]
[[module.imports]]
path = "github.com/google/docsy"
[[module.imports]]
path = "github.com/google/docsy/dependencies"
```

**Note**: you could also alter the theme line to: 

```
theme = "github.com/google/docsy"
```

Both are valid (and more or less equivalent) solutions.

dominikh reviewed (commented):
Thanks, but I don't think that Hugo modules will work for us. In particular the interaction with `go mod tidy` seems problematic. There is `hugo mod tidy`, but that only considers the modules imported by Hugo, not any of our other Go code in this module.

Hugo modules seem to be rather new, and I'd rather not become an early adopter. Our current system works better for us.

dominikh commented:
Thanks; I'll take another look at this in a couple days.

KarthikNayak commented:
@deining are you still working on this?

deining commented:
> @deining are you still working on this?

Please keep that issue open. I hope to find time soon to update the @docsy@ submodule only.

deining commented:
I updated this PR to just bump the theme submodule to latest docsy version and removed your modifications that are not needed any more after this update.

dominikh commented:
Merged and published, thanks.
